src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -511,41 +462,37 @@ def getCEStatus(self):
     def _renewJobs(self, jobList):
         """Written for the REST interface - jobList is already in the REST format
         This function is called only by this class, NOT by the SiteDirector"""
-
-        # Get the new proxy once for all the jobs ...
-        newProxy = X509Chain()
-        res = newProxy.loadProxyFromFile(self.session.cert)
-
-        # Renew the jobs
         for job in jobList:
             # First get the delegation (proxy)
             dID = self._delegation(job)
```

Comment:
Hi @aldbr 

About the delegations, which CE did you test against? For me, it worked with lancaster

ce = "grendel.hec.lancs.ac.uk"

However, I did not have luck with any of the RAL CEs. So, it depends on the version of the ARC CE used I suppose.

About the job renewal, I am not sure I understand. Is your suggestion to simply renew all the delegations at the given CE? This may mean renewing delegations without any jobs, which are best left to expire ...

Reply:
> About the job renewal, I am not sure I understand. Is your suggestion to simply renew all the delegations at the given CE? This may mean renewing delegations without any jobs, which are best left to expire ...

Let's say jobs A and B are submitted at the same time, and C is submitted later. We would have:
- delegation1 is bound to jobs A, B
- delegation2 is bound to job C

Your method does:
- get delegation of A: delegation1; renew it
- get delegation of B: delegation1; renew it
- get delegation of C: delegation2; renew it

In this context, delegation1 is renewed twice.

Whereas you could first make a list of delegations. You would have:
- get delegation of A: delegation1
- get delegation of B: delegation1
- get delegation of C: delegation2
- list of delegations [delegation1, delegation2]
- renew delegation1, delegation2

Do I understand correctly the code?
What do you think?

Reply:
Good point. I was trying to avoid having two loops. This also will work I agree - when you have some time, will you be able to provide a patch, or shall I do it?

It is not something that is going to happen often. So, I am happy for testing this module to go ahead with the code as it exists and update this in the next iteration.